app-connector-client/test/test_integration.js

```diff
@@ -1,115 +1,189 @@
+#!/usr/bin/env node
+'use strict'
 
 var request = require('supertest');
-var server = require("../server/app")("test/varkes_config.json");
+const mock = require('../server/app')
+const express = require("express")
 const fs = require("fs")
 const path = require("path")
+const kyma = require("@varkes/example-kyma-mock")
 const serviceMetadata = path.resolve("test/service-metadata.json")
 
-if (process.env.NODE_ENV == "test") {
-    require("../server/keys").generatePrivateKey()
+const port = 10001 //! listen in different port
+const tokenURL = `http://localhost:${port}/connector/v1/applications/signingRequests/info?token=123`
+
+describe("should work", () => {
+    var kymaServer
+    var server
+    before(async () => { //* start kyma mock before tests
+
+        await deleteKeysFile()
+
+        await kyma.then(app => {
+            kymaServer = app.listen(port)
+        })
+        await mock("./test/varkes_config.json").then((mock) => {
+            server = express()
+            server.use(mock)
+        })
+
+        await request(server)  //* Make sure we are connected to kyma
+            .post("/connection").send(
+
+                { "url": tokenURL }
+
+            ).set('Accept', 'application/json').
+            expect(200)
+
+    })
+
+    after(() => { //* stop kyma mock after tests
+        kymaServer.close()
+        deleteKeysFile()
+    })
 
     describe('basic routes', function () {
-        it('responds to /', function testSlash(done) {
-            request(server)
+        it('responds to /', function testSlash() {
+            return request(server)

```

Comment:
why you changed the approach to return a value always instead of calling done callback?